src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax;
+
+import core.basesyntax.fileoperations.FileReader;
+import core.basesyntax.fileoperations.FileReaderImpl;
+import core.basesyntax.fileoperations.FileWriter;
+import core.basesyntax.fileoperations.FileWriterImpl;
+import core.basesyntax.fileoperations.ReportGenerator;
+import core.basesyntax.fileoperations.ReportGeneratorImpl;
+import core.basesyntax.operations.BalanceOperation;
+import core.basesyntax.operations.OperationHandler;
+import core.basesyntax.operations.OperationStrategy;
+import core.basesyntax.operations.OperationStrategyImpl;
+import core.basesyntax.operations.PurchaseOperation;
+import core.basesyntax.operations.ReturnOperation;
+import core.basesyntax.operations.SupplyOperation;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Main {
+    public static void main(String[] arg) {
+        // 1. Read the data from the input CSV file
+        FileReader fileReader = new FileReaderImpl();
+        List<String> inputReport = fileReader.read("reportToRead.csv");
+
+        // 2. Convert the incoming data into FruitTransactions list
+        DataConverter dataConverter = new DataConverterImpl();
+        final List<FruitTransaction> transactions = dataConverter.convertToTransaction(inputReport);
+
+        // 3. Create and feel the map with all OperationHandler implementations
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlers = new HashMap<>();
+        operationHandlers.put(FruitTransaction.Operation.BALANCE, new BalanceOperation());
+        operationHandlers.put(FruitTransaction.Operation.PURCHASE, new PurchaseOperation());
+        operationHandlers.put(FruitTransaction.Operation.RETURN, new ReturnOperation());
+        operationHandlers.put(FruitTransaction.Operation.SUPPLY, new SupplyOperation());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationHandlers);
+
+        // 4. Process the incoming transactions with applicable OperationHandler implementations
+        ShopService shopService = new ShopServiceImpl(operationStrategy);
+        shopService.process(transactions);
+
+        // 5.Generate report based on the current Storage state
+        ReportGenerator reportGenerator = new ReportGeneratorImpl();
```

Comment:
The `ReportGenerator` should be independent as well. It's better to call its methods in the `main()` method rather than creating a dependency between the `ShopService` and `ReportGenerator`.